modules/components/src/DataTable/TableToolbar/helpers.js

```diff
@@ -0,0 +1,67 @@
+import download from '../../utils/download';
+
+export const saveTSV = async ({ url, files = [], fileName, options = {} }) =>
+  download({
+    url,
+    method: 'POST',
+    ...options,
+    params: {
+      fileName,
+      files: files.map(({ allColumns, columns, exColumns = null, ...file }, i) => ({
+        ...file,
+        columns: exColumns // if the component gave you custom columns to show
+          ? Object.values(
+              exColumns.length > 0 // if they ask for any specific columns
+                ? exColumns.map((fieldName) => allColumns[fieldName]) // use them
+                : allColumns, // else, they're asking for all the columns
+            )
+          : columns.filter((column) => column.show), // no custom columns, use admin's
+      })),
+      ...options.params,
+    },
+  });
+
+export const exporterProcessor = (exporter, allowTSVExport, exportTSVText) => {
+  const exporterArray =
+    Array.isArray(exporter) &&
+    exporter.map((item) =>
+      [item, item?.function].some((fnName) => fnName === 'saveTSV') ||
+      (item.constructor === {}.constructor && !Object.keys(item).includes('function'))
+        ? {
+            exLabel: item?.label || exportTSVText,
+            exFunction: saveTSV,
+            ...(item?.columns && Array.isArray(item.columns) && { exColumns: item?.columns }),
+          }
+        : Object.entries(item).reduce(
+            (exItem, [key, value]) => ({
+              ...exItem,
+              [`ex${key[0].toUpperCase()}${key.slice(1)}`]: value,
+            }),
+            {},
+          ),
+    );
+
+  const multipleExporters = exporterArray && exporter.length > 1;
+
+  // this checks whether a single custom function has been provided
+  // by itself, or as the single item in an array
+  const customExporter =
+    exporter && //
+    (exporter instanceof Function // check it's a valid function
```

Comment:
This ternary chain is particularly tough to read, need to clean this up to make it easier to read and work with.

Because you are trying to assign the value, using an anonymous method could be useful. Here's one potential implementation using a switch for all the cases instead of an if/else ladder or ternary chain.

```
const resolveCustomExporter = () => {
    switch(true) {
        case exporter instanceof Function:
            return exporter;
        case exporterArray && exporter.length > 1:
            return exporterArray;
        case exporterArray && exporter.length === 1;
            return exporterArray[0]?.exFunction;
        default:
            console.error('...');
    }
}
```

Reply:
Gotta be honest: I'm a big fan of nested ternaries, because I usually indent them in what I find is a very readable way.
Prettier here works against that style, so your suggestion to use a switch is of course perfectly adequate.
Will amend this now. Thanks for the suggestion!